### DISCUSSION TITLE
Impact of CPU numbers

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#23764] [Q&A General] [answered]
Hello,

I faced a weird problem. When I run my simulation on my linux system and on hpc with 8 CPUs, my solution will converge.

But when I increase the number of CPUs to 28 and 94, my simulation will diverge.

I wanted to know whether this is a common problem and how I can resolve the convergence problem with increasing number of CPUs.

Thank you,
Ali

**Comment by GiudGiud** [comment, marked as answer]
Hello

The preconditioning is affected by the number of CPUs. You should look at designing preconditioning that is more resilient in parallel.
What does your preconditioning block look right now?

Guillaume

**Comment by alimostafavi24** [comment]
```
[Preconditioning]
  [./SMP_PJFNK]
    type = SMP
    full = true
    solve_type = Newton
  [../]
[]

[Executioner]
  type = Transient
  petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_levels'
  petsc_options_value = 'bjacobi  ilu          4'

  nl_rel_tol  = 1e-3
  nl_abs_tol  = 1e-3
  nl_max_its  = 70
  l_tol       = 1e-3
  l_max_its   = 70

  #end_time    = 200

 # dt = 0.001
 end_time = 10


#adaptive time stepping
[./TimeStepper]
   type = IterationAdaptiveDT
	    dt = 0.0005
 [../]
[]
```

**Reply by GiudGiud** [reply]
I think the block in block jacobi depend on the partition of the problem.
Maybe try to use the overlap parameters to sooth this?
